elastic/indexer/elastic-indexer.go

```diff
@@ -52,20 +55,117 @@ func New(esHosts []string, esIndex, esMapping, kafkaAddress, changesTopic, kafka
-	m, err := i.kReader.FetchMessage(context.Background())
+	messages, err := i.readMessageBlock(5*time.Second, i.bulkChunkSize)
+	log.Println("Messages Bulk: ", len(messages))
+	if err != nil {
+		return err
+	}
+
+	if len(messages) == 0 {
+		return nil
+	}
+
+	var bulkBody string
+	bulkDocumentIdKafkaMessages := make(map[string]kafka.Message)
+	for _, message := range messages {
+
+		changeMessage := &changestream.ChangeMessage{}
+		if err := json.Unmarshal(message.Value, changeMessage); err != nil {
+			return err
+		}
+		bulkOperation, err := i.indexFunc(changeMessage)
+
+		if err != nil {
+			return err
+		}
+
+		bulkDocumentIdKafkaMessages[bulkOperation.DocumentId] = message
+		bulkBody += bulkOperation.BulkOperation
+	}
+
+	bulkResponse, err := i.esClient.Bulk(strings.NewReader(bulkBody), i.esClient.Bulk.WithIndex(i.esIndex))
+	if err != nil {
+		return err
+	}
+	log.Println("Result Messages Bulk: ", bulkResponse.Status())
+
+	body, err := ioutil.ReadAll(bulkResponse.Body)
+
 	if err != nil {
 		return err
 	}
 
-	changeMessage := &changestream.ChangeMessage{}
-	if err := json.Unmarshal(m.Value, changeMessage); err != nil {
+	var result bulkResult
+	err = json.Unmarshal(body, &result)
+
+	if err != nil {
 		return err
 	}
 
-	if err := i.indexFunc(i.esClient, changeMessage); err != nil {
+	log.Println("BulkResultItem: ", len(result.Items))
+	log.Println("BulkResult: ", string(body))
+
+	err = i.checkAllResultMessagesAreValid(&result)
+	if err != nil {
 		return err
 	}
-	return i.kReader.CommitMessages(context.Background(), m)
+
+	for _, bulkResultOperation := range result.Items {
+
+		if bulkResultOperation.Index.ID != "" {
+			err := i.commitKafkaMessages(bulkResultOperation.Index.Status, bulkResultOperation.Index.ID, bulkDocumentIdKafkaMessages)
+			if err != nil {
+				return err
+			}
+		} else if bulkResultOperation.Update.ID != "" {
+			err := i.commitKafkaMessages(bulkResultOperation.Update.Status, bulkResultOperation.Update.ID, bulkDocumentIdKafkaMessages)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (i *Indexer) checkAllResultMessagesAreValid(result *bulkResult) error {
+	if result == nil {
+		return fmt.Errorf("BulkResult is nil")
+	}
+	for _, bulkResultOperation := range result.Items {
+		if bulkResultOperation.Index.ID != "" {
+
+			err := errorForHttpStatus(bulkResultOperation.Index.Status)
+			if err != nil {
+				return err
+			}
+		} else if bulkResultOperation.Update.ID != "" {
+
+			err := errorForHttpStatus(bulkResultOperation.Index.Status)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func errorForHttpStatus(httpStatus int) error {
+	if httpStatus != 200 && httpStatus != 201 {
+		return fmt.Errorf("creating/updateing index failed Httpstatus: %d \n", httpStatus)
+	}
+	return nil
+}
+
+func (i *Indexer) commitKafkaMessages(httpStatus int, documentId string, bulkDocumentIdKafkaMessages map[string]kafka.Message) error {
+	if httpStatus == 200 || httpStatus == 201 {
```

Comment:
We don't need to check the httpStatus anymore, right? 
When  `commitKafkaMessages` is called, the items have already been checked 
-> I don't think we need a helper func for commiting the messages anymore :)

Reply:
haha that's true  :)